runtime/compiler/optimizer/EscapeAnalysisTools.cpp

```diff
@@ -62,44 +62,84 @@ void TR_EscapeAnalysisTools::insertFakeEscapeForOSR(TR::Block *block, TR::Node *
    int32_t byteCodeIndex = bci.getByteCodeIndex();
    TR_OSRCompilationData *osrCompilationData = _comp->getOSRCompilationData();
 
+   TR_OSRMethodData *osrMethodData  = _comp->getOSRCompilationData()->getOSRMethodDataArray()[inlinedIndex + 1];
```

Comment:
Can you please add a comment to explain why we need the map and how this analysis will not be sound, in general, without it. Since we have the env var to revert people may wonder/forget why it is there and it is an easily overlooked problem.

Reply:
Added a comment in commit https://github.com/eclipse/openj9/pull/9802/commits/76b9e2c7de823623969f22afd69aa3b234ed4c26